### PR TITLE
Use SecondBase in a static way

### PR DESCRIPTION
**Pull request by neumayer**

### CODE REVIEW

File: example/src/main/java/org/secondbase/example/main/HelloStatic.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.secondbase.example.main;
```
Comment (acidmoose) [inline example/src/main/java/org/secondbase/example/main/HelloStatic.java]: This class can probably be removed? It should be covered by other examples.

File: core/src/main/java/org/secondbase/core/SecondBase.java [hunk 2]
```diff
@@ -11,35 +9,12 @@
...
-     * Set up SecondBase.
-     * @param args command line arguments
-     * @param flags preloaded Flags class
-     */
-    public SecondBase(final String[] args, final Flags flags) throws SecondBaseException {
-        init(args, flags);
-    }
-
-    private void init(final String[] args, final Flags flags) throws SecondBaseException {
+    public SecondBase(final String[] args, final SecondBaseModule[] modules, final Flags flags) throws SecondBaseException {
```
Comment (acidmoose) [inline core/src/main/java/org/secondbase/core/SecondBase.java]: long line

File: example/src/main/java/org/secondbase/example/main/HelloJsonLogging.java [hunk 3]
```diff
@@ -12,7 +14,8 @@ public static void main(final String[] args) throws SecondBaseException {
                 "--service", "test",
                 "--datacenter", "dc1"
         };
-        new SecondBase(realArgs, new Flags().loadOpts(helloJsonLogging));
+        final SecondBaseModule[] modules = new SecondBaseModule[] { new JsonLoggerModule() };
+        new SecondBase(realArgs, modules, new Flags().loadOpts(helloJsonLogging));
```
Comment (acidmoose) [inline example/src/main/java/org/secondbase/example/main/HelloJsonLogging.java]: Examples that doesn't revolve around the use of Flags shouldn't have flags in them. Can we remove the flags in this class and add the Flag-less SecondBase constructor again?

SecondBase(final SecondBaseModule[] modules)
SecondBase(final SecondBaseModule[] modules, final Flags flags)

**Review by acidmoose — CHANGES_REQUESTED** [review]

File: example/src/main/java/org/secondbase/example/main/HelloStatic.java (hunk 1 above)
Comment (neumayer) [reply example/src/main/java/org/secondbase/example/main/HelloStatic.java]: Done.

File: core/src/main/java/org/secondbase/core/SecondBase.java (hunk 2 above)
Comment (neumayer) [reply core/src/main/java/org/secondbase/core/SecondBase.java]: Done.

File: example/src/main/java/org/secondbase/example/main/HelloJsonLogging.java (hunk 3 above)
Comment (neumayer) [reply example/src/main/java/org/secondbase/example/main/HelloJsonLogging.java]: I think so.

File: core/src/main/java/org/secondbase/core/SecondBase.java [hunk 4]
```diff
@@ -1,45 +1,26 @@
...
+    private Flags flags;
 
-    /**
-     * Set up SecondBase.
-     * @param args command line arguments
-     * @param flags preloaded Flags class
-     */
-    public SecondBase(final String[] args, final Flags flags) throws SecondBaseException {
-        init(args, flags);
+    public SecondBase(final SecondBaseModule[] modules) throws SecondBaseException{
```
Comment (acidmoose) [inline core/src/main/java/org/secondbase/core/SecondBase.java]: Sorry, I mistyped the constructor example. There is never any case where the user will not use args in any way. Either he has his own Flags or he has modules which has flags. So I think we should have these two constructors:

SecondBase(final String[] args, final SecondBaseModule[] modules)
SecondBase(final String[] args, final SecondBaseModule[] modules, final Flags flags)

Comment (neumayer) [reply core/src/main/java/org/secondbase/core/SecondBase.java]: I'm glad. I did as you said again.

**Review by acidmoose — APPROVED** [review]

**Pull request merged**